packages/grid/x-data-grid/src/hooks/features/editRows/useGridRowEditing.new.ts

```diff
@@ -677,7 +680,7 @@ export const useGridRowEditing = (
 
     // Update the ref here because updateStateToStopRowEditMode may change it later
     const copyOfPrevRowModesModel = prevRowModesModel.current;
-    prevRowModesModel.current = rowModesModel;
+    prevRowModesModel.current = deepClone(rowModesModel); // Do a deep-clone because the attributes might be changed later
```

Comment:
It's not part of the main solution but I noticed we just store the prop in a ref then later change this ref, which also changes the value of the prop. Doing a deep clone ensures that it won't leak.